src/ir/utils.h

```diff
@@ -109,7 +109,9 @@ struct ReFinalize
   // block finalization is O(bad) if we do each block by itself, so do it in
   // bulk, tracking break value types so we just do a linear pass
 
-  std::map<Name, Type> breakValues;
+  // TODO: Switch to std::unordered_set once types are properly canonicalized so
+  // determinism isn't an issue.
```

Comment:
why not do the change now? i'm not sure what's preventing it.

Reply:
If we have an unordered_set, the stable sort in `ensureSuperType` is no better than an unstable sort. If we have two equivalent-but-distinct types A and B that are supertypes of all the other types being considered, then whether A or B is chosen will be nondeterministic. Using a std::map uses < rather than == to check uniqueness, so only A or B will be sent to `ensureSuperType`, not both.

Reply:
Oh right, I missed that part...

Another option could be an unordered set that remembers the insertion order. That wouldn't need to wait on anything.